src/pages/cart/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { useState, useEffect } from 'react';
+
+import { getProductDetail } from '@/api/products';
+import { Button } from '@/components/cart/Button';
+import { CartItem } from '@/components/cart/CartItem';
+import { EmptyCart } from '@/components/cart/EmptyCart';
+import { Header } from '@/components/cart/Header';
+import { PaymentButton } from '@/components/cart/PaymentButton';
+import { CartLoading } from '@/components/loading/CartLoading';
+import { getCartItems, deleteCartItem, updateCartItem } from '@/utils/cart';
+import { formattedPrice, calculateTotalPrice } from '@/utils/price';
+
+export default function CartPage() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [productsInCart, setProductsInCart] = useState<
+    (Product & { quantity: number })[]
+  >([]);
+  const [checkList, setCheckList] = useState<boolean[]>([]);
+  const [totalPrice, setTotalPrice] = useState(0);
+
+  const handleCheck = (index: number) => {
+    const updatedCheckList = [...checkList];
+    updatedCheckList[index] = !updatedCheckList[index];
+    setCheckList(updatedCheckList);
+  };
+
+  const handleQuantityChange = (index: number, newQuantity: number) => {
+    const updatedProducts = [...productsInCart];
+    updatedProducts[index].quantity = newQuantity;
+    setProductsInCart(updatedProducts);
+    updateCartItem(updatedProducts[index].id, newQuantity);
+  };
```

Comment:
Suggest: Array를 처리할때는 만들때부터 이렇게 처리하면 어떤가요?
```suggestion
const handleQuantityChange = (index: number, newQuantity: number) => {
  const updatedProducts = productsInCart.map((product, i) =>
    i === index ? { ...product, quantity: newQuantity } : product
  );
  setProductsInCart(updatedProducts);
  updateCartItem(updatedProducts[index].id, newQuantity);
};
```